src/dpi.rs

```diff
@@ -1,80 +1,102 @@
+//! will truncate the fractional part of the float, rather than round to the nearest integer - the
+//! proper behavior. Use the provided `cast` function or `From`/`Into` conversions, which handle the
+//! rounding properly. Note that precision loss will still occur when rounding from a float to an
+//! int, although rounding lessens the problem.
 //!
-//! Your GPU has no awareness of the concept of logical pixels, and unless you like wasting pixel density, your
-//! framebuffer's size should be in physical pixels.
+//! ### Events
 //!
-//! `winit` will send [`Resized`](../enum.WindowEvent.html#variant.Resized) events whenever a window's logical size
-//! changes, and [`HiDpiFactorChanged`](../enum.WindowEvent.html#variant.HiDpiFactorChanged) events
-//! whenever the DPI factor changes. Receiving either of these events means that the physical size of your window has
-//! changed, and you should recompute it using the latest values you received for each. If the logical size and the
-//! DPI factor change simultaneously, `winit` will send both events together; thus, it's recommended to buffer
-//! these events and process them at the end of the queue.
+//! Winit will dispatch a [`HiDpiFactorChanged`](crate::event::WindowEvent::HiDpiFactorChanged)
+//! event whenever a window's scale factor has changed. This can happen if the user drags their
+//! window from a standard-resolution monitor to a high-DPI monitor, or if the user changes their
+//! DPI settings. This gives you a chance to rescale your application's UI elements and adjust how
+//! the platform changes the window's size to reflect the new scale factor. If a window hasn't
+//! received a [`HiDpiFactorChanged`](../enum.WindowEvent.html#variant.HiDpiFactorChanged) event,
+//! then its scale factor is `1.0`.
 //!
-//! If you never received any [`HiDpiFactorChanged`](../enum.WindowEvent.html#variant.HiDpiFactorChanged) events,
-//! then your window's DPI factor is 1.
+//! ## How is the scale factor calculated?
+//!
+//! Scale factor is calculated differently on different platforms:
+//!
+//! - **Windows:** On Windows 8 and 10, per-monitor scaling is readily configured by users from the
+//!   display settings. While users are free to select any option they want, they're only given a
+//!   selection of "nice" scale factors, i.e. 1.0, 1.25, 1.5... on Windows 7, the scale factor is
+//!   global and changing it requires logging out. See [this article][windows_1] for technical
+//!   details.
+//! - **macOS:** "retina displays" have a scale factor of 2.0. Otherwise, the scale factor is 1.0.
+//!   Intermediate scale factors are never used. It's possible for any display to use that 2.0 scale
+//!   factor, given the use of the command line.
+//! - **X11:** Many man-hours have been spent trying to figure out how to handle DPI in X11. Winit
+//!   currently uses a three-pronged approach.
+//!   + Calcuate the scale factor based on the millimeter monitor dimensions provided by XRandR.
```

Comment:
Why this odd reverse ordering? Why not something like "Winit first looks for the value set in the `WINIT_X11_SCALE_FACTOR` environment variable, using it if present. If not present, Winit looks for [...]"

Reply:
There was a reason, but I'm too tired to articulate what it was right now. I've updated the docs to hopefully be better-worded.